### PR TITLE
fix: removed check for cpu templates for aarch64

### PR DESCRIPTION
**Pull request by ShadowCurse**
## Changes

Removed check for cpu templates for aarch64.
This is done, because now we have static templates for aarch64 and we forgot to remove this restriction earlier.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/integration_tests/functional/test_api.py [hunk 1]
```diff
@@ -411,12 +411,12 @@ def test_api_machine_config(test_microvm_with_api):
...
 
     # Test that CPU template errors on ARM.
-    response = test_microvm.machine_cfg.patch(cpu_template="C3")
     if platform.machine() == "x86_64":
+        response = test_microvm.machine_cfg.patch(cpu_template="C3")
         assert test_microvm.api_session.is_status_no_content(response.status_code)
     else:
-        assert test_microvm.api_session.is_status_bad_request(response.status_code)
-        assert "CPU templates are not supported on aarch64" in response.text
+        response = test_microvm.machine_cfg.patch(cpu_template="None")
```
Comment (kalyazin) [inline tests/integration_tests/functional/test_api.py:418]: What does it actually test? How is it different from not passing the argument?

**Review by kalyazin — COMMENTED** [review]
Could you add some more info to the PR/commit message? What is expected/actual behaviour?

File: tests/integration_tests/functional/test_api.py [hunk 2]
```diff
@@ -411,12 +411,12 @@ def test_api_machine_config(test_microvm_with_api):
         )
 
     # Test that CPU template errors on ARM.
```
[the remark this answers is not in the document]
Comment (ShadowCurse) [reply tests/integration_tests/functional/test_api.py]: removed the whole section

File: tests/integration_tests/functional/test_api.py (hunk 1 above)
Comment (ShadowCurse) [reply tests/integration_tests/functional/test_api.py]: it is. it is not `None` but `"None"`

Comment (ShadowCurse) [reply tests/integration_tests/functional/test_api.py]: and `"None"` is not `None`

Comment (kalyazin) [reply tests/integration_tests/functional/test_api.py]: I guess it deserves some comment explaining that we check that the "None" template is supported.

**Comment by JonathanWoollett-Light** [comment]
Nit: [`coverage: updated test coverage`](https://github.com/firecracker-microvm/firecracker/pull/3628/commits/2c54b6fa2a42a508bf9083b7e76ab58bf2580878) could be renamed [`test: updated test coverage`](https://github.com/firecracker-microvm/firecracker/pull/3628/commits/2c54b6fa2a42a508bf9083b7e76ab58bf2580878)

File: tests/integration_tests/functional/test_api.py (hunk 1 above)
Comment (ShadowCurse) [reply tests/integration_tests/functional/test_api.py]: added comment

**Review by zulinx86 — APPROVED** [review]

**Pull request merged**